rosbag2_py/src/rosbag2_py/_transport.cpp

```diff
@@ -190,16 +190,23 @@ class Player
 class Recorder
 {
 private:
-  std::unique_ptr<rclcpp::executors::SingleThreadedExecutor> exec_;
+  static std::unique_ptr<rclcpp::executors::SingleThreadedExecutor> exec_;
```

Comment:
Why is this member static now? Because of the `rclcpp::init/shutdown` in here, it doesn't  really make sense to have multiple `rclpy::Recorder` instances, but the semantics of this global variable don't really make sense to me. Would these be better as just a global variable, or a singleton object?

Reply:
@emersonknapp Hmm, I didn't think about the situation when multiple instances of the `rclpy::Recorder` and one static executor. This is a good point.
I made the executor static to be able to call its static `exec_->cancel();` method from signal handlers to interrupt execution.  
In the case of the situation when multiple instances of the `rclpy::Recorder` exists it looks reasonable I would say even must have to make the executor as static to convey signal interruption event to all instances. 

However, I agree that it would be better to wrap the static executor in the singleton pattern. I am sorry I missed it, - will fix.
Good catch!